your-rents-data-jooq/src/main/java/com/yourrents/data/jooq/daos/PropertyDao.java

```diff
@@ -0,0 +1,69 @@
+package com.yourrents.data.jooq.daos;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import com.yourrents.data.jooq.tables.records.PropertyRecord;
+
+import static com.yourrents.data.jooq.Tables.PROPERTY;
+
+/*-
+ * #%L
+ * Your Rents Data JOOQ
+ * %%
+ * Copyright (C) 2019 - 2021 Your Rents Team
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import org.jooq.DSLContext;
+import org.jooq.SortField;
+
+public class PropertyDao {
+
+    private DSLContext dsl;
+
+    /**
+     * Create a new PropertyDao with an attached configuration
+     */
+    public PropertyDao(DSLContext dsl) {
```

Comment:
no lombok? 😞

Reply:
Non mi piace avere lombok per cose così semplici. Inoltre non c'è come dipendenza in data-jooq, e appunto mi sembra un po' overkill introdurla solo per un costruttore di un unico parametro. Anche in core lo trovo spesso un po' forzato, utile per set/get e builder, ma per i costruttori si risolverebbe quasi sempre con un'annotazione @Autowired o simili (tanto lì Spring c'è per forza). Prossimamente vorrei passare a java 17, e anche per la faccenda dto probabilmente sarà più appropriato l'uso di record Java. Boh, vedremo.